### PR TITLE
Economy!

### PR DESCRIPTION
**Pull request by theOperand**
probably doesnt compile yet on account of me not having added the config options for currency yet.

### CODE REVIEW

**Comment by Miauw62** [comment]
>tfw it actually compiles

WOT

**Comment by pudl** [comment]
tbh i was working on this too... lol... guess u got it first

**Comment by Miauw62** [comment]
xhuis also has a pr for this you butte

**Comment by xxalpha** [comment]
Bad and not necessary. Goes against the play style of tg servers. :-1:

Also why two prs for this?

**Comment by Xhuis** [comment]
PFFFFFFFF

**Comment by freerealestate** [comment]
Can you explain what this does for people who can't read code very well?

**Comment by Miauw62** [comment]
You can read the general idea in this thread: https://tgstation13.org/phpBB/viewtopic.php?f=10&t=3993
The code that is currently in place is a simple system with a central "bank" datum that can be used to look up accounts by their number and a bank account datum with a balance and logs. There are some functions in place to withdraw from or deposit to a bank account, as well as a function to transfer money from one account to another. However, this is not implemented in any way in the game world yet.

**Comment by Jalleo** [comment]
I like this PR as it is it merge post haste :+1: tbh its starting there was talk for it years now we better make sure to not repeat the other codebases mistakes

**Comment by ike709** [comment]
:+1: woo we can have department budgets and stuff

Also, maybe Miauw and Xhuis should both continue working on their own PRs and we can merge the better one. The battle for capitalism begins.

**Comment by Summoners** [comment]
:+1: Would love to have economy, even if the /tg/ servers don't want to use it I still would love to see it in code for future uses.

**Comment by Fox-McCloud** [comment]
If no RP Goon can have a functional, fun, and interesting economy...pretty sure TG can as well as it has some RP.

:+1:  for this--really interested to see how this will develop!

**Comment by Jalleo** [comment]
Rev could cause and then protest about a lack of wages for everyone. Viva la burning cash.

**Comment by Miauw62** [comment]
That's it for today, quite a bit of work on ATMs, UIs are a real PITA to do.

Banking servers wouldn't really require much more effort since they don't really do much by themselves, they only really do something if they're blown up :P

**Comment by Miauw62** [comment]
there, bankserver!

**Comment by Xhuis** [comment]
hey you caught up :^)

File: code/modules/economy/atm.dm [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+		updateDialog()
+		return
+	
+	if(href_list["search"])
+		var/tosearch = href_list["tosearch"]
+		updateDialog()
+		return
+
+	if(href_list["depositcash"])
+		var/value = 0
```
Comment (Miauw62) [inline code/modules/economy/atm.dm]: i should probably put these in their own procs lest i invoke the wrath of gia's ghost

**Comment by Miauw62** [comment]
economy is not habbeding ;_;7

**Pull request closed without merging**